src/ovs/aca_ovs_l2_programmer.cpp

```diff
@@ -189,12 +188,7 @@ int ACA_OVS_L2_Programmer::configure_port(const string vpc_id, const string port
   // internal vlan id or to create a new vpc_id entry to get a new internal vlan id
   int internal_vlan_id = ACA_Vlan_Manager::get_instance().get_or_create_vlan_id(vpc_id);
 
-  ACA_Vlan_Manager::get_instance().add_ovs_port(vpc_id, port_name);
-
-  execute_openflow_command(
-          "add-flow br-tun \"table=4, priority=1,tun_id=" + to_string(tunnel_id) +
-                  " actions=mod_vlan_vid:" + to_string(internal_vlan_id) + ",output:\"patch-int\"\"",
-          culminative_time, overall_rc);
+  ACA_Vlan_Manager::get_instance().create_ovs_port(vpc_id, port_name, tunnel_id, culminative_time);
 
   if (g_demo_mode) {
     string cmd_string = "add-port br-int " + port_name +
```

Comment:
My comment is to Line 194 where we start to execute the ovsdb command and system command. 

This is one of the place we can start abstracting more dp related logics, for example, we have a one dp interface class with a run method, and many worker classes inheriting from this interface class with the real programming logics.

One step further, we could have one more layer of abstraction "on top of" the interface class to represent more general dp programming.

Reply:
yes, all the logic of parsing port configuration is already done at this point, and ready to do the dp programming. Another abstraction layer we can do is at line 191 ACA_Vlan_Manager::get_instance().create_ovs_port. If the other dp behavior is similar to ovs, we can simply call into another dataplane specific code here.

Reply:
sounds good. Let us plan for that and create a tracking item.